Colton feedback (#30821)
dagster-io/dagster: docs/docs/etl-pipeline-tutorial/3-data-quality.md

sidebar_position: 40
---

Data quality is critical in data pipelines. Inspecting individual assets ensures that data quality issues are caught before they affect the entire pipeline. In this step, you will:

- Define an asset check
- Execute that asset check in the UI

## 1. Define an asset check

> In Dagster, you define [asset checks](/guides/test/asset-checks) similarly to how you define assets. Asset checks run when an asset is materialized and ensure that certain criteria are met.

Reviewer: ```suggestion
In Dagster, you define [asset checks](/guides/test/asset-checks) the same way you define assets. Asset checks run when an asset is materialized and ensure that certain criteria are met.
```

> With the Dagster UI, you can now see that an asset check is associated with the `joined_data` asset.

Reviewer: ```suggestion
In the Dagster UI, you can now see that an asset check is associated with the `joined_data` asset:
```

Reviewer: When was the `joined_data` asset introduced? Maybe part of the transform step is missing?

![2048 resolution](/images/tutorial/etl-tutorial/asset-check.png)

Asset checks will run when an asset is materialized, but asset checks can also be executed manually in the UI:

> 1. Reload your Definitions.

Reviewer: I think we will need to tell people to run `dg dev` again and navigate to `127.0.0.1:3000` here
